[WIP] Add new semantic markup support to antsibull-docs

felixfontein opened the pull request:
Implements antsibull-docs part of ansible/ansible#73137. See also ansible/ansible#74937.

~~(For now contains #272 so I have less chaos locally :) )~~

webknjaz commented on src/antsibull/lint_extra_docs.py:
```diff
@@ -46,6 +50,12 @@ def lint_optional_conditions(content: str, path: str, collection_name: str
 
 
+def _setup_rstcheck():
+    '''Make sure that rstcheck knows about our roles.'''
```
TIL. I'd usually put `rst-roles` in `.flake8` but that doesn't scale.

webknjaz commented on src/sphinx_antsibull_ext/roles.py:
```diff
@@ -0,0 +1,71 @@
+Add roles for semantic markup.
+'''
+
+from __future__ import (absolute_import, division, print_function)
```
Does this code have to work on python 2?

felixfontein replied:
No, but the other files in this directory also have it.

felixfontein replied:
And gone it is :)

felixfontein commented:
I updated this PR to the latest specification in https://hackmd.io/VjN60QSoRSSeRfvGmOH1lQ?both, with the small difference that `[]` can be used in option names and return value names, and there is a `missing:` prefix (as opposed to `FQCN#type:`) to indicate that this is not an existing option/return value name (for example to reference something that has been removed in a previous release). I've also updated https://github.com/ansible-community/antsibull/pull/373 and re-built my docsite (https://ansible.fontein.de) with this. The references are now also linking to the option/return value documentation. Examples:
- https://ansible.fontein.de/collections/community/dns/docsite/hosttech_guide.html#querying-dns-records-and-record-sets (note the links in the sentence below the first example)
- https://ansible.fontein.de/collections/community/dns/docsite/hosttech_guide.html#bulk-synchronization-of-dns-record-sets (note the links in the sentence below the first example; also shows the `[]` notation)
- https://ansible.fontein.de/collections/community/dns/hetzner_dns_record_module.html#parameters (note that `record`, `zone_name`, and `prefix` can be clicked)
- https://ansible.fontein.de/collections/community/dns/hetzner_dns_record_sets_module.html#parameter-record_sets (shows `[]` and `.` notation)
- https://ansible.fontein.de/collections/community/dns/wait_for_txt_module.html#parameter-records (shows `[]` and `.` notation)

(Currently includes #373.)

felixfontein commented:
This will be continued in https://github.com/ansible-community/antsibull-docs/pull/4 since antsibull-docs is removed from this package.
